api.c

```diff
@@ -846,72 +913,72 @@ int tcmu_emulate_mode_sense(
 	int i;
 	int ret;
 	size_t used_len;
-	uint8_t buf[512];
-	bool got_sense = false;
+	uint8_t *buf;
+        uint8_t *orig_buf = NULL;
```

Comment:
Giving myself a review comment. Use tab here instead of spaces.

Also for a alloc_len of 0 I think I should have returned an ILLEGAL REQUEST instead of returning success.